BraveRewardsUI/Wallet/Notifications/NotificationHandler.swift

```diff
@@ -0,0 +1,192 @@
+  func hideNotifications()
+}
+
+class NotificationHandler {
+
+  private var ledger: BraveLedger
+  weak var delegate: NotificationHandlerDelegate?
+  
+  init(ledger: BraveLedger) {
+    self.ledger = ledger
+  }
+  
+  deinit {
+    stop()
+  }
+  
+  weak var currrentNotification: RewardsNotification?
+  
+  func start() {
+    // Add observer
+    stop()
+    NotificationCenter.default.addObserver(self, selector: #selector(notificationAdded(_:)), name: .init(rawValue: "BATBraveLedgerNotificationAdded"), object: nil)
+    loadNotification()
+  }
+  
+  private func stop() {
+    NotificationCenter.default.removeObserver(self, name: .init(rawValue: "BATBraveLedgerNotificationAdded"), object: nil)
+  }
+  
+  @objc private func notificationAdded(_ notification: Notification) {
+    // TODO: Filter notification?
+    if currrentNotification == nil {
+      loadNotification()
+    }
+  }
+  
+  private func loadNotification() {
+    guard delegate != nil else {
+      return
+    }
+    if let currrentNotification = currrentNotification {
+      ledger.clearNotification(currrentNotification)
+      self.currrentNotification = nil
+    }
+    
+    //.failedContribution, .invalid & .backupWallet are not supported right now
+    // TODO: Verify that notifications are ordered & they are displayed in FIFO.
+    let ignoredRewardNotificationTypes: [RewardsNotification.Kind] = [.failedContribution,
+                                                                      .invalid,
+                                                                      .backupWallet]
+    let notification = ledger.notifications
+    if let notification = notification.first(where: {
+      !ignoredRewardNotificationTypes.contains($0.kind)
+    }) {
+      currrentNotification = notification
+      switch notification.kind {
+      case .autoContribute, .tipsProcessed, .verifiedPublisher:
+        addMessageNotification(notification: notification)
+      case .grant, .grantAds, .insufficientFunds, .pendingNotEnoughFunds, .adsLaunch:
+        addActionNotification(notification: notification)
+      default:
+        return
+      }
+    } else {
+      delegate?.hideNotifications()
+    }
+  }
+  
+  private func addActionNotification(notification: RewardsNotification) {
+    let category: WalletActionNotification.Category
+    let body: String
+    
+    switch notification.kind {
+    case .grant:
+      body = Strings.NotificationGrantNotification
+      category = .grant
+    case .grantAds:
+      body = Strings.NotificationEarningsClaimDefault
+      category = .grant
+    case .insufficientFunds:
+      body = Strings.NotificationInsufficientFunds
+      category = .insufficientFunds
+    case .pendingNotEnoughFunds:
+      body = Strings.NotificationPendingNotEnoughFunds
+      category = .insufficientFunds
+    case .adsLaunch:
+      body = Strings.NotificationBraveAdsLaunchMsg
+      category = .adsLaunch
+    default:
+      loadNotification()
```

Comment:
Not sure what this is here for, isn't this undefined behaviour? Shouldn't we just `assertionFaliure`?